javagen/src/main/java/com/azure/autorest/template/EnumTemplate.java

```diff
@@ -47,6 +49,7 @@ private void writeExpandableStringEnum(EnumType enumType, JavaFile javaFile, Jav
         if (!settings.isStreamStyleSerialization()) {
             imports.add("com.fasterxml.jackson.annotation.JsonCreator");
         }
+        addGeneratedImport(imports);
```

Comment:
We should do the same for `writeEnum` method too in this class. We should add `@Generated` annotation to enum values and any public methods on the enum type.

Reply:
ok, I will add to enums as well. (Previously I thought this is to support models, so I only changes enums under `models` package. I did not change for enums.)

Reply:
I assume Java `Enum` also generated under models package?

Though I've currently no idea how we are going to customize the `Enum` (but guess adding it won't hurt).

Reply:
Sorry I did not say it clearly, enums are generated in `models` package, but I did not see the usage in DPG/TypeSpec. We are using enums for `ServiceVersion`, it is not under model, so I did not add Generated annotation previously.

Reply:
Usage is that `@fixed` over tsp enum. We should have such case in both cadl-ranch test and local test.

We may add some method to the `Enum`, e.g. `getValueAsInt` that convert String on wire to Int to customer. But this is a very hypothesis use case.